src/Nix/Exec.hs

```diff
@@ -336,7 +336,7 @@ execBinaryOp scope span op lval rarg = do
             _                  -> nverr $ ErrorCall $ unsupportedTypes lval rval
 
         (NVStr ls, NVStr rs) -> case op of
-            NPlus -> pure $ bin nvStrP (ls `mappend` rs)
+            NPlus -> pure $ bin nvStrP (ls `hackyStringMappend` rs)
```

Comment:
Can you explain a bit more of the motivation behind this?

Reply:
As long as at some point in the future we go back to using `Monoid` if this thing turns out to be a monoid, I'm fine with interim measures.